plugins/rapid7_insightvm/komand_rapid7_insightvm/triggers/scan_completion/trigger.py

```diff
@@ -145,25 +143,29 @@ def query_results_from_latest_scan(scan_id: int) -> str:
         """
 
         return (
-            f"SELECT fasvi.scan_id, fasvi.asset_id, fasvi.vulnerability_id, dv.cvss_v3_score, dvr.source, daga.asset_group_id, dss.solution_id, dss.summary, dv.nexpose_id "  # nosec B608
+            f"WITH matching_asset_group_ids AS (SELECT asset_id, string_agg(CAST(asset_group_id AS varchar), ',') AS asset_group_ids "  # nosec B608
+            f"FROM dim_asset_group_asset "
+            f"GROUP BY asset_id) "
+            f"SELECT fasvi.scan_id, fasvi.asset_id, fasvi.vulnerability_id, dv.nexpose_id, dv.severity, dv.cvss_v3_score, dvc.category_name, ds.solution_id, ds.summary, dvr.source "
             f"FROM fact_asset_scan_vulnerability_instance AS fasvi "
-            f"JOIN dim_asset_group_asset AS daga ON (fasvi.asset_id = daga.asset_id) "
-            f"JOIN dim_vulnerability AS dv ON (fasvi.vulnerability_id = dv.vulnerability_id) "
-            f"JOIN dim_vulnerability_reference AS dvr ON (fasvi.vulnerability_id = dvr.vulnerability_id) "
-            f"JOIN dim_solution AS dss ON (dv.nexpose_id = dss.nexpose_id) "
+            f"INNER JOIN dim_vulnerability AS dv ON (fasvi.vulnerability_id = dv.vulnerability_id) "
+            f"INNER JOIN dim_vulnerability_category AS dvc ON (fasvi.vulnerability_id = dvc.vulnerability_id) "
+            f"INNER JOIN dim_solution AS ds ON (dv.nexpose_id = ds.nexpose_id) "
+            f"INNER JOIN matching_asset_group_ids AS magi ON (fasvi.asset_id = magi.asset_id) "
+            f"LEFT JOIN dim_vulnerability_reference AS dvr ON (fasvi.vulnerability_id = dvr.vulnerability_id) "
             f"WHERE fasvi.scan_id = {scan_id} "
+            f"GROUP BY fasvi.scan_id, fasvi.asset_id, fasvi.vulnerability_id, magi.asset_group_ids, dv.nexpose_id, dv.cvss_v3_score, dvc.category_name, ds.solution_id, ds.summary, dvr.source, dv.severity "
```

Comment:
I think this can be handled as. multiline f-string 
```
f"""
WITH matching....
"""
```

Reply:
Do you mind if I keep it single line now for readability and to match how it is also done in the `top_remediations` action queries?

Reply:
I think it's more readable in that case. Also, you are currently using f-string while not passing any variables to it. What do you think about updating it to `f""""""`? 😄